src/Types/Type.php

```diff
@@ -106,7 +106,10 @@ abstract public function getSQLDeclaration(array $column, AbstractPlatform $plat
      *
      * @todo Needed?
      */
-    abstract public function getName();
+    public function getName()
+    {
+        return self::getTypeRegistry()->lookupName($this);
```

Comment:
The introduction of a dependency of a value on a registry doesn't look right, even temporarily. If we want to deprecate and eventually get rid of `Type::getName()`, we should do this.

In its current state, this pull request removes some code at a cost of introducing a risk of BC break.

Instead, what if we should identify the scenarios that rely on `Type::getName()` and deprecate them. For instance, deprecate registering a type with a name different than the value returned by `Type::getName()`.

Reply:
> The introduction of a dependency of a value on a registry doesn't look right, even temporarily. If we want to deprecate and eventually get rid of Type::getName(), we should do this.

I don't understand this paragraph. What value are you referring to? Did you mean to write "we should _not_ do this."?

Reply:
Sorry for the confusion. I was referring to the return value of `Type::getName()`.

> Did you mean to write "we should not do this."?

I meant that if we want to deprecate and eventually get rid of `Type::getName()`, we should do _that_ (provide the upgrade path and deprecate it).

> If we were to drop `Type::getName()`, we would probably replace calls to it by `Type::getTypeRegistry()->lookupName($type)`, wouldn't we?

If we have a type registered under a name, we can use the type name in the API and look up the type by name. Why do we need to do the reverse lookup? What does it even mean given that the types will be no longer flyweights?

Reply:
> If we have a type registered under a name, we can use the type name in the API and look up the type by name. Why do we need to do the reverse lookup?

Well for instance here: https://github.com/doctrine/dbal/blob/ebd986b6e4f205b54cc27f5d5ba0323d84fa29b1/src/Platforms/AbstractPlatform.php#L445-L454

we have access to the type instance, but not to its name in the registry, so we would have to change the signature of this method to also provide the name, which will be a BC-break, and we will also need to change the calling methods if it does not have access to the name, etc. etc.

Or maybe we should use `spl_object_id` to identify the `Type` instance in that particular scenario, I don't know. I guess we would have to check for each scenario if we have an alternate why of fulfilling it.

Reply:
Why can't we just call `$type->requiresSQLCommentHint($this)`?